dspace-uclouvain/src/main/java/org/dspace/uclouvain/pdfAttestationGenerator/xmlworkflow/actions/SendEmailAttestationAction.java

```diff
@@ -0,0 +1,270 @@
+     * @throws Exception
+     */
+    private void sendPromoterEmail(HashMap<String, List<String>> metadata, Item dspaceItem, String templatePath, InputStream attestation) throws Exception {
+        Email email = this.generateBaseEmail(metadata, templatePath, metadata.get("advisors_email"), attestation);
+        this.appendUrlsToEmail(metadata, email, dspaceItem);
+        email.send(); 
+    }
+
+    /**
+     * Send an error email to both the submitters and the promoters with the stacktrace of the exception.
+     * This is done when an error is occurring when generating the PDF attestation (or a mail).
+     * @param metadata: A HashMap containing all the metadata of the submission.
+     * @param submitter: The EPerson who submitted the item.
+     * @param dspaceItem: The DSpace item corresponding to the submission.
+     * @param e: The exception that occurred.
+     * @throws Exception
+     */
+    private void sendErrorEmail(HashMap<String, List<String>> metadata, EPerson submitter, Item dspaceItem, Exception e) throws Exception {
+        Email email = Email.getEmail(this.source + "/config/emails/pdf_attestation_error");
+        // TODO: Use given emails
+        // for (String address : metadata.get("authors_email")) {
+        //     email.addRecipient(address);
+        // }
+        // for (String address : metadata.get("advisors_email")) {
+        //     email.addRecipient(address);
+        // }
+        email.addRecipient("michael.pourbaix@uclouvain.be");
+        email.addArgument(LocalDateTime.now().format(DateTimeFormatter.ofPattern("dd/MM/yyyy - HH:mm:ss")));
+        email.addArgument(submitter.getFullName());
+        email.addArgument(metadata.get("dspace_entity_type").get(0).toLowerCase());
+        email.addArgument("Here are some information that might be useful for our team:\n -> Item's uuid: " + dspaceItem.getID() + "\n-> Stacktrace:\n" + ExceptionUtils.getStackTrace(e));
+        email.setSubject(this.mailErrorSubject);
+
+        email.send();
+    }
+
+    /**
+     * Generates a resume for the given submission containing the title, the abstract and the authors.
+     * 
+     * @param map The HashMap containing information about the submission.
+     * @return The resume as a String.
+    */
+    private static String generateSubmissionResume(HashMap<String, List<String>> map) throws ResumeGenerationException {
+        try {
+            BinaryOperator<String> parser = (subtotal, element) -> subtotal + element + "; ";
+
+            String title = map.get("dc_title").get(0);
+            String authors = map.get("dc_contributor_author").stream().reduce("", parser);
+            String promoters = map.get("dc_contributor_advisor").stream().reduce("", parser);
+            String abstractText = map.get("dc_description_abstract").get(0);
+            
+            return "Title: " + title + "\nAuthor(s): " + authors + "\nPromoters: " + promoters + "\nAbstract: " + abstractText;
```

Comment:
You have problem if one attribute is "null" (it shouldn't... but at UCLouvain everything is possible)
Add tests to prevent this possibility.